crates/router/src/core/payments.rs

```diff
@@ -375,29 +376,46 @@ where
             should_continue_capture,
         );
 
-        operation
-            .to_domain()?
-            .call_external_three_ds_authentication_if_eligible(
-                state,
-                &mut payment_data,
-                &mut should_continue_transaction,
-                &connector_details,
-                &business_profile,
-                &key_store,
-                mandate_type,
-            )
-            .await?;
-        operation
-            .to_domain()?
-            .call_unified_authentication_service_if_eligible(
-                state,
-                &mut payment_data,
-                &mut should_continue_transaction,
-                &connector_details,
-                &business_profile,
-                &key_store,
-            )
-            .await?;
+        if helpers::is_merchant_eligible_authenthention_service(merchant_account.get_id(), state)
+            .await?
+        {
+            let authentication_product_ids = business_profile
```

Comment:
why can't we have this inside call_unified_authentication_service_if_eligible? We don't want to spill any login in payments operation core ideally!